sdk/communication/azure-communication-phonenumbers/test/_shared/testcase.py

```diff
@@ -25,14 +25,14 @@ def process_response(self, response):
                     if isinstance(item, str):
                         body["phoneNumbers"] = [self._replacement]
                         break
-                    elif "phoneNumber" in item:
+                    if "phoneNumber" in item:
                         item['phoneNumber'] = self._replacement
-                    elif "id" in item:
+                    if "id" in item:
                         item['id'] = self._replacement
             response['body']['string'] = json.dumps(body)
             response['url'] = self._replacement
             return response
-        except (KeyError, ValueError):
+        except (KeyError, ValueError, TypeError):
```

Comment:
why we need this change for the User-agent change?

Reply:
This is a fix for the type error issue we have in the search available phone numbers test. We weren't catching that exception before. It's totally unrelated to the user agent issue, though.